### PR TITLE
FEM-2390 add PlaybackCaptions support ovp/ott

### PR DESCRIPTION
**Pull request by giladna**
## need to add boolean useApiCaptions in the ovp provider init parameters  that tells the provider if to build the sbbs alone

### CODE REVIEW

**Comment by giladna** [comment]
"playbackCaptions": [{
				"label": "English",
				"format": "1",
				"language": "English",
				"webVttUrl": "http:\/\/cdnbakmi.kaltura.com\/api_v3\/index.php\/service\/caption_captionasset\/action\/serveWebVTT\/captionAssetId\/0_6y2kppi8\/segmentIndex\/-1\/version\/2\/captions.vtt",
				"url": "http:\/\/cdnbakmi.kaltura.com\/api_v3\/index.php\/service\/caption_captionAsset\/action\/serve\/captionAssetId\/0_6y2kppi8\/v\/2",
				"isDefault": true,
				"languageCode": "en",
				"objectType": "KalturaCaptionPlaybackPluginData"
			}
		],

need to build this from the response automatically

List<PKExternalSubtitle> subtitleList = new ArrayList<>();
PKExternalSubtitle pkExternalSubtitleEn = new PKExternalSubtitle()
.setUrl("https://mkvtoolnix.download/samples/vsshort-en.srt")
.setMimeType(PKSubtitleFormat.srt)
.setLabel("External_English")
.setLanguage("eng")
.setDefault();
subtitleList.add(pkExternalSubtitleEn);
mediaEntry.setExternalSubtitleList(subtitleList);

**Review by GouravSna — APPROVED** [review]

File: mediaproviders/src/main/java/com/kaltura/playkit/providers/ovp/KalturaOvpMediaProvider.java [hunk 1]
```diff
@@ -361,13 +367,40 @@ public static PKMediaEntry getMediaEntry(String baseUrl, String ks, String partn
...
+
+            for (KalturaPlaybackCaption kalturaPlaybackCaption : playbackCaptionList) {
+
+                String subtitleURL = kalturaPlaybackCaption.getWebVttUrl();
+
+                PKExternalSubtitle pkExternalSubtitle = new PKExternalSubtitle()
+                        .setUrl(subtitleURL)
+                        .setMimeType(PKSubtitleFormat.valueOfUrl(subtitleURL))
+                        .setLabel(kalturaPlaybackCaption.getLabel())
+                        .setLanguage(kalturaPlaybackCaption.getLanguage());
```
Comment (OrenMe) [inline mediaproviders/src/main/java/com/kaltura/playkit/providers/ovp/KalturaOvpMediaProvider.java]: use `languageCode` instead - this is the field which is used to populate the manifests as well

File: mediaproviders/src/main/java/com/kaltura/playkit/providers/api/ovp/model/KalturaPlaybackCaption.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    public String getUrl() {
+        return url;
+    }
+
+    public boolean isDefault() {
+        return isDefault;
+    }
+}
+
+//"playbackCaptions": [{
```
Comment (OrenMe) [inline mediaproviders/src/main/java/com/kaltura/playkit/providers/api/ovp/model/KalturaPlaybackCaption.java]: Either use this as javadoc or remove

**Review by OrenMe — CHANGES_REQUESTED** [review]

File: mediaproviders/src/main/java/com/kaltura/playkit/providers/ovp/KalturaOvpMediaProvider.java (hunk 1 above)
Comment (giladna) [reply mediaproviders/src/main/java/com/kaltura/playkit/providers/ovp/KalturaOvpMediaProvider.java]: done

File: mediaproviders/src/main/java/com/kaltura/playkit/providers/api/ovp/model/KalturaPlaybackCaption.java (hunk 2 above)
Comment (giladna) [reply mediaproviders/src/main/java/com/kaltura/playkit/providers/api/ovp/model/KalturaPlaybackCaption.java]: done

**Review by OrenMe — APPROVED** [review]

**Pull request merged**